jerry-core/vm/vm.c

```diff
@@ -625,11 +625,11 @@ vm_init_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
 
             ecma_object_t *ref_base_lex_env_p = ecma_op_resolve_reference_base (frame_ctx_p->lex_env_p, name_p);
 
-            /* TODO: check the return value */
-            ecma_op_put_value_lex_env_base (ref_base_lex_env_p,
-                                            name_p,
-                                            is_strict,
-                                            lit_value);
+            ecma_value_t put_value_result = ecma_op_put_value_lex_env_base (ref_base_lex_env_p,
+                                                                            name_p,
+                                                                            is_strict,
+                                                                            lit_value);
+            ecma_free_value (put_value_result);
```

Comment:
As I see, based on the standards, this function will call the 'ecma_op_set_mutable_binding' method (in this case base must be a reference whose base is an environment record), and because this will be an attempt to change the value of an immutable binding, and the strict param is also true, the binding cannot be set and throw a TypeError exception.